File: GFG POTD/018-May-2025/024-sum-of-all-substrings-of-a-number.java
// https://www.geeksforgeeks.org/problems/sum-of-all-substrings-of-a-number-1587115621/1




class Solution {
    public static int sumSubstrings(String s) {
        int n = s.length();
        long result = 0;
        long prev   = 0;
        for (int i = 0; i < n; i ++) {
            int num = s.charAt(i) - '0';
            prev = prev * 10 + (num * (i + 1));
            result += prev;
        }
        return (int)result;
    }
}

File: GFG POTD/018-May-2025/024-sum-of-all-substrings-of-a-number.cpp
// https://www.geeksforgeeks.org/problems/sum-of-all-substrings-of-a-number-1587115621/1




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N)
    // SC : O(1)
    int sumSubstrings(string &s) {
        long long result = 0;
        long long prev   = 0;
        for (int i = 0; i < s.length(); i ++) {
            int num = s[i] - '0';
            prev = prev * 10 + (num * (i + 1));
            result += prev;
        }
        return result;
    }
};